backend/src/services/requestHandlers.ts

```diff
@@ -0,0 +1,48 @@
+import { IncomingMessage, ServerResponse } from 'http';
+
+type GetCallback<T> = {
+    (req: IncomingMessage, reqUrl: URL): Promise<T>;
+};
+
+type PostCallback<T> = {
+    (postedData: T): Promise<T>;
+};
+
+export function postHandler<T>(
+    req: IncomingMessage,
+    res: ServerResponse,
+    callback: PostCallback<T>
+) {
+    req.setEncoding("utf8");
+    let data = "";
+    req.on("data", (chunk: string) => {
+        data += chunk;
+    });
+    req.on("end", async () => {
+        const postedData = JSON.parse(data) as T;
+        const postResponse = await callback(postedData);
+        res.end(JSON.stringify(postResponse));
+    });
+}
+
+export async function getHandler<T>(
+    req: IncomingMessage,
+    res: ServerResponse,
+    reqUrl: URL,
+    callback: GetCallback<T>
+) {
+    const getResponse = await callback(req, reqUrl);
+    if (getResponse) {
+        res.writeHead(200);
+        res.end(JSON.stringify(getResponse));
+    } else {
+        res.writeHead(404);
+        res.end();
+    }
+    res.end();
```

Comment:
I would move all of this response-centric stuff inside the specific controllers and let them decide how to respond. Some controller actions might want to return a 202 or something, for example

Reply:
By parameterising the response code maybe? To avoid repetition I could still use these functions right? Or do you suggest not using these functions at all

Reply:
ah actually this is for get and there isn't that much logic here that can be reused.. was thinking more for the Post handling, but I saw your comment below

Reply:
Yeah, I'd just get rid of them